algorithms/max-flow/max-flow.tex

```diff
@@ -0,0 +1,82 @@
+\section{Max Flows}\index{graph!Max Flows}
+#ifdef hackpackpp
+There are several types of problems that involve finding maximum flow in a graph.
+Many of these problems can easily tackled by making small transformations to the graph.
+
+
+If there is more than one source or sink, one can add a "super" source and or sink that connects to all of the sources and sinks.
+Then find the flow from the super source to the super sink.
+
+
+If the graph is undirected, simply insert every edge twice in both directions.
+
+
+If the flow is limited through the nodes instead of the edges, simply spit every node into a "in" node and an "out" node.
+
+
+\begin{itemize}
+	\item While there is a path from source to sink
+	\begin{itemize}
+		\item Greedily find the widest path
+		\item Reduse the cost of the nodes on the widest path by its width
+		\item Create reverse paths by the width of the widest path
+	\end{itemize}
+\end{itemize}
+
+In a directed acyclic graph, this can be done in linear time.
+
+\begin{itemize}
+	\item Topologically sort the graph using a breadth first search
+	\item Use dynamic programming to build up the maximum flow.
+\end{itemize}
+#endif
+
+\subsection{Applications}
+\begin{itemize}
+	\item Finding the maximum flow in a graph
+	\item Finding a minimum set of edges required to disconnect source and sink
+	\item Finding a maximal matching
+\end{itemize}
+#ifdef hackpackpp
+
+\subsection{Example Contest Problem: Cow-Ex}
+The cows are up to another of the insidious schemes.
+The cows have setup a milk distribution system on Farmer John's farm.
+
+The pipes that the cows use to transport the milk are of various sizes.
+Each pipe has been rated with a positive integer referring to the amount of milk that the pipe can carry in 1 minute.
+
+The cows want to know how much milk they can transport to the barn at once.
+\subsubsection{Sample Format}
```

Comment:
Throughout the hackpack, this section is called "Input", "Input Format", or now "Sample Format". For clarity and standardization, the hack pack should only use one.